### ISSUE TITLE
Support for 2018.4.2f1

### ISSUE THREAD

**Issue by 036** [vfsfitvnm/frida-il2cpp-bridge#8]
I have an Android app I'm trying to reverse engineer but it's using Unity 2018.4.2f1 as confirmed by il2cppinspector. I'm assuming there won't be any support for this?

![image](https://user-images.githubusercontent.com/17926309/122818352-f9bee580-d2d0-11eb-9e43-2d1dc098b32e.png)

![image](https://user-images.githubusercontent.com/17926309/122818286-e875d900-d2d0-11eb-9d9a-c5ce46c0f2f0.png)

I tried bypassing the issue by setting the Unity version to 2018.3.8 in variables.ts and rebuilding but I get the below error

![image](https://user-images.githubusercontent.com/17926309/122818578-4a364300-d2d1-11eb-874d-f34137001a1e.png)

**Comment by vfsfitvnm** [comment]
@036 
Hi, thanks for this issue - it never happened to me.
I gather the unity version here: https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/variables.ts#L51.
That error means it couldn't find the hardcoded string, so `unityVersion` remains undefined.

Would you help me track this down?
Also,
- Which platform are you using? (Windows/Android)
- Which application are you dealing with? (If you can say so)

**Comment by vfsfitvnm** [comment]
@036 
Hi, thanks for this issue - it never happened to me.
I gather the unity version here: https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/variables.ts#L51.
That error means it couldn't find the hardcoded string, so unityVersion remains undefined.

Would you help me track this down?
Also,

 -  Which platform are you using? (Windows/Android)
 -  Which application are you dealing with? (If you can say so)

**Comment by 036** [comment]
Hey, thanks for looking into this for me. 

I did rebuild the module after editing the `GetUnityVersion` function to return `2018.3.8 `  however I then ran into the problem of 'not a function' I believe this was coming from `variables_1.unityVersion.isEqualOrAbove is not a function`.

I'm running this on Android and the APK is _reserved_.

Happy to have a discussion somewhere if that helps.

**Comment by vfsfitvnm** [comment]
@036 
So, if you want to rebuild the module, you can edit `getUnityVersion` to make it return a `UnityVersion`:
```ts
async function getUnityVersion() {
    return new UnityVersion("2018.4.2f1");
}
```
Just tried and works.
I'm going to figure out what's the problem with the automatic version search.

**Comment by 036** [comment]
Thanks @vfsfitvnm will try it out now!

**Comment by vfsfitvnm** [comment]
@036 
Another update: I tried another app (100% used to work), and it doesn't work anymore. Frida's `Module.enumerateRanges()` doesn't find any memory range anymore, when used on unity's shared libraries.
I tried an older app version and worked, so something has changed, definitely.

Shared libraries doesn't get extracted to `/data/app/<package>/lib/<abi>` anymore, but remains in the apk (e.g. `/data/app/<package>/split_config.<abi>.apk!/lib/<abi>/`). In this scenario, Frida's `enumerateRanges`, `enumerateImpots`, `enumerateExports`, `enumerateRanges` won't work anymore (but `findExportByName` or `getExportByName` will).
